src/explore-education-statistics-common/src/components/form/FormCheckboxSearchGroup.tsx

```diff
@@ -117,7 +117,7 @@ const FormCheckboxSearchGroup = ({
           value={value}
           name={name}
           options={filteredOptions}
-          searchOnly
+          searchOnly={searchOnly}
```

Comment:
Only just looked at these search changes and got a few issues with what we've done here.

`BaseFormCheckboxGroup` is only supposed to be a simple component that handles the rendering of checkboxes. Search is something that should be handled by`FormCheckboxSearchGroup` (it's in the name).

From what I can see, it seems strange that we opted to mix both of these concerns into `BaseFormCheckboxGroup` and would call this a smell.

Would appreciate some insight into the thought process here, or otherwise, I'd recommend that we pull all of this logic out of `BaseFormCheckboxGroup` and into `FormCheckboxSearchGroup` and refactor everything accordingly.